pkg/wallet/wallet.go

```diff
@@ -63,12 +62,59 @@ func (w *BaseWallet) DecryptMessage(encMessage []byte, toVerKey string) ([]byte,
 
 // PackMessage Pack a message for one or more recipients.
 func (w *BaseWallet) PackMessage(envelope *Envelope) ([]byte, error) {
-	return nil, fmt.Errorf("not implemented")
+	if envelope == nil {
+		return nil, errors.New("envelope argument is nil")
+	}
+	// get keypair from db
+	senderKeyPair, err := w.getKey(envelope.FromVerKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed from getKey: %w", err)
+	}
+
+	var recipients []*[crypto.KeySize]byte
+	for _, verKey := range envelope.ToVerKeys {
+		// decode base58 ver key
+		verKeyBytes := base58.Decode(verKey)
+		// create 32 byte key
+		var recipient [crypto.KeySize]byte
+		copy(recipient[:], verKeyBytes)
+		recipients = append(recipients, &recipient)
+	}
+
+	// encrypt message
+	bytes, err := w.crypter.Encrypt(envelope.Message,
+		crypto.KeyPair{Priv: senderKeyPair.Priv, Pub: senderKeyPair.Pub}, recipients)
```

Comment:
second argument can simply be senderKeyPair, no need to create a new KeyPair instance

Reply:
done